In [ ]:
# Frank is 20 years old

age = 20
name = 'Frank'
# print(name + ' was ' + str(age) + ' years old')
# print('%s was %d years old' % (name, age))
# print('{} was {} years old'.format(name, age))
# print('{name} was {age} years old'.format(name=name, age=age))
print(f'{name} was {age} years old')

- 占位符
    - `%d` 整数
    - `%f` 浮点数
    - `%s` 字符串
    - `%x` 16进制

In [ ]:
print('%.2f' % (100.123456))
print('%.3f' % (100.123456))
print('%+.3f' % (100.123456))
print('%-.3f' % (100.123456))
print('%+.3f' % (-100.123456))
print('%-.3f' % (-100.123456))
print('start%10.3fend' % (100.123456))
print('start%-10.3fend' % (100.123456))

In [ ]:
"{} {}".format("hello", "world")    # 不设置指定位置，按默认顺序

In [ ]:
"{0} {1}".format("hello", "world")  # 设置指定位置

In [ ]:
"{1} {0} {1}".format("hello", "world")  # 设置指定位置

In [ ]:
print("网站名：{name}, 地址 {url}".format(name="baidu", url="www.baidu.com"))
 
# 通过字典设置参数
site = {"name": "baidu", "url": "www.baidu.com"}
print("网站名：{name}, 地址 {url}".format(**site))
 
# 通过列表索引设置参数
my_list = ['baidu', 'www.baidu.com']
print("网站名：{0[0]}, 地址 {0[1]}".format(my_list))  # "0" 是必须的

In [33]:
print("{:.2f}".format(3.1415926));
print("{:+.2f}".format(3.1415926));
print("|{:10.2f}|".format(3.1415926));
print("|{:>10.2f}|".format(3.1415926));
print("|{:<10.2f}|".format(3.1415926));
print("|{:^10.2f}|".format(3.1415926));
print("|{:0>10.2f}|".format(3.1415926));
print("|{:*<10.2f}|".format(3.1415926));

3.14
+3.14
|      3.14|
|      3.14|
|3.14      |
|   3.14   |
|0000003.14|
|3.14******|


- `^`, `<`, `>` 分别是居中、左对齐、右对齐，后面带宽度
- `:` 号后面带填充的字符，只能是一个字符，不指定则默认是用空格填充
- `+` 表示在正数前显示 +，负数前显示 -
- ` `（空格）表示在正数前加空格

- examples

数字            格式         输出            描述
3.1415926      {:.2f}      '3.14'          保留小数点后两位
3.1415926      {:+.2f}     '+3.14'         带符号保留小数点后两位
-1             {:+.2f}     '-1.00'         带符号保留小数点后两位
2.71828        {:.0f}      '3'             不带小数
5              {:0>2d}     '05'            数字补零 (填充左边, 宽度为2)
5              {:x<4d}     '5xxx'          数字补x (填充右边, 宽度为4)
10             {:x<4d}     '10xx'          数字补x (填充右边, 宽度为4)
1000000        {:,}        '1,000,000'     以逗号分隔的数字格式
0.25           {:.2%}      '25.00%'        百分比格式
1000000000     {:.2e}      '1.00e+09'      指数记法
13             {:10d}      '        13'    右对齐 (默认, 宽度为10)
13             {:<10d}     '13        '    左对齐 (宽度为10)
13             {:^10d}     '    13    '    中间对齐 (宽度为10)

- `b`、`d`、`o`、`x` 分别是二进制、十进制、八进制、十六进制

In [34]:
print('{:b}'.format(11))
print('{:d}'.format(11))
print('{:o}'.format(11))
print('{:x}'.format(11))
print('{:#x}'.format(11))
print('{:#X}'.format(11))

1011
11
13
b
0xb
0XB


In [35]:
print ("{} 对应的位置是 {{0}}".format("baidu"))

baidu 对应的位置是 {0}
